crates/ethereum/payload/src/lib.rs

```diff
@@ -543,7 +551,8 @@ where
     let sealed_block = block.seal_slow();
     debug!(target: "payload_builder", ?sealed_block, "sealed built block");
 
-    let mut payload = EthBuiltPayload::new(attributes.id, sealed_block, total_fees);
+    let receipts_pay: Vec<Receipt> = receipts.iter().flatten().cloned().collect();
```

Comment:
since we don't use `receipts` again after this, we can do `into_iter` and avoid the clone